libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/ContraIndicators.java

```diff
@@ -27,20 +27,32 @@ public Integer getContraIndicatorScore(
     }
 
     public Optional<ContraIndicator> getLatestContraIndicator() {
-        return contraIndicatorsMap.values().stream()
-                .max(Comparator.comparing(ContraIndicator::getIssuanceDate));
+        // An overly complicated way to ensure a deterministic result if there are multiple CIs
+        // with the same issue instant - which will probably never happen anyway.
+        return usersContraIndicators.stream()
+                .max(
+                        Comparator.comparing(ContraIndicator::getIssuanceDate)
+                                .thenComparing(
+                                        (a, b) ->
+                                                String.format("%s%s", a.getCode(), a.getDocument())
+                                                        .compareToIgnoreCase(
+                                                                String.format(
+                                                                        "%s%s",
+                                                                        b.getCode(),
+                                                                        b.getDocument()))));
```

Comment:
Is this equivalent to:
```
return usersContraIndicators.stream()
                .max(
                        Comparator.comparing(ContraIndicator::getIssuanceDate)
                                .thenComparing(ContraIndicator::getCode)
                                .thenComparing(ContraIndicator::getDocument));
```
Which I think conveys the intent more clearly?

Reply:
haha. yeah, much better